### PR TITLE
Search for nodes now reflects how it works in game

### PR DESCRIPTION
**Pull request by eldemonstro**
How it works in game:
- Multiple keywords separated by name (eg: `Phys crit` will show all nodes that have `phys` and `crit` somewhere in the title and description)
- Quotation marks (`"`) can be used to search for the exact text (eg: `"Phys crit"` will not highlight any node but `"Sword Damage and Critical Strike Multiplier"` will)

How it works on Path of Building:
- It searches the exact text everytime (reflecting quotation marks in game)

What this PR tend to achieve:
- Copy the way that it is used in the game to create a "consistency" between both searchs

### CODE REVIEW

**Review by ppoelzl — CHANGES_REQUESTED** [review]
The formatting of the PR seems to be all over the place. Please fix it.

**Comment by eldemonstro** [comment]
Reformatted the code, seems like a tab vs spaces problem on my side, sorry

**Review by ppoelzl — COMMENTED** [review]
It's very hard to follow your code and it doesn't match this project's standards. I'm sorry, but I can't accept it in this state and you will most likely have to completely redo it for it to get accepted. Feel free to give it another try after you become more familiar with our project though.
